src/commonMain/kotlin/org/hisp/dhis/rules/models/RuleVariableCurrentEvent.kt

```diff
@@ -1,34 +1,31 @@
 package org.hisp.dhis.rules.models
 
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.toLocalDateTime
 import org.hisp.dhis.rules.engine.RuleVariableValue
-import org.hisp.dhis.rules.utils.getLastUpdateDate
 
 class RuleVariableCurrentEvent(
     override val name: String,
     override val useCodeForOptionSet: Boolean,
     override val options: List<Option>,
     override val field: String,
-    override val fieldType: RuleValueType,
+    override val fieldType: RuleValueType
 ) : RuleVariable {
     override fun createValues(
         ruleEvent: RuleEvent?,
-        allEventValues: Map<String, List<RuleDataValue>>,
+        allEventValues: Map<String, List<RuleDataValueHistory>>,
         currentEnrollmentValues: Map<String, RuleAttributeValue>,
         currentEventValues: Map<String, RuleDataValue>
-    ): Map<String, RuleVariableValue> {
-        val valueMap: MutableMap<String, RuleVariableValue> = HashMap()
-        val variableValue: RuleVariableValue
-        variableValue = if (currentEventValues.containsKey(field)) {
+    ): RuleVariableValue {
+        val variableValue = if (currentEventValues.containsKey(field)) {
             val value = currentEventValues[field]
             val optionValue = if (useCodeForOptionSet) value!!.value else getOptionName(value!!.value)!!
             RuleVariableValue(
                 fieldType, optionValue,
-                listOf(optionValue), getLastUpdateDate(listOf(value))
-            )
+                listOf(optionValue), ruleEvent!!.eventDate.toLocalDateTime(TimeZone.currentSystemDefault()).date.toString())
```

Comment:
Just a comment, not necessarily something to change now. In general, I will try to avoid the use of `!!` because it just throws a NPE without any other information. Something such as 
`requireNotNull(ruleEvent) { "Event must not be null in CurrentEvent variables" }` 
might be more meaningful.